CS56-StoreMapper/StoreMapper
# Add route visualization enhancements and UI improvements

## Description

This PR introduces several enhancements to the route visualization and overall user interface:

- Implement color-coded route segments based on speed limits
- Add a legend control to explain route color coding
- Create a route information control displaying distance and estimated time
- Update route distance calculation to use miles instead of kilometers
- Improve route segment data structure for better frontend rendering
- Add Enter key functionality to search input for easier use
- Implement HTML entity decoding for location names
- Update README with new UI screenshot
- Minor code cleanup and optimization

Testing:
- Manually tested route creation and visualization
- Verified legend and route information controls appear correctly
- Checked Enter key functionality for search
- Confirmed correct display of location names with special characters

Note: Some tests have been temporarily disabled and will need to be updated in a future PR.
